common/collision.go

```diff
@@ -139,6 +148,12 @@ type collisionEntity struct {
 	*SpaceComponent
 }
 
+type Collisioner interface {
```

Comment:
These are good thoughts.  I did change styles half way through as I was doing it, partly to see which I prefered.  I've been using it for a month now, and I think I know how it can be done better.

I want to get rid of the "Get"  part  of Get__Component etc , but the problem is, the parent object will hold the object and inherit the method, so I can't call it the same name as the object as this will cause confusion possibly to the compiler and definitly to some programmers.

I'm going to re-edit and make a new pull request shortly